pkg/grpc_middleware/dialer/config.go

```diff
@@ -0,0 +1,196 @@
+package dialer
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/caring/go-packages/v2/pkg/errors"
+)
+
+// Config read from environment or connection builder.
+type Config struct {
+	host       string
```

Comment:
should this be called ConnStringInfo? it seems it's only holding parsed values from the connection string...

also, do we even need this, or can we just add method to the connection builder that parses from the connection string and sets all these? What is the value of this struct as an intermediary holding ground for this data?

Reply:
A refactor to merge the two together makes sense.

Reply:
Thinking about this more there is a fair amount of complexity in parsing the connection address and building the TLS Config that does not relate directly to the Builder that makes sense to keep it separate.